config/201-clusterrole.yaml

```diff
@@ -87,6 +87,8 @@ rules:
   - leases
   verbs:
   - create
+  - get
+  - update
```

Comment:
Those are already defined below, at line 97 and 98.

Looking at the error message, it seems like you are running the **RabbitMQ** event source, but submitted a PR for the **GitLab* source (this repository).

Reply:
Ops, wrong copy paste. You can find below the correct logs from my installation. 
```
E0322 22:32:53.916254       1 leaderelection.go:361] Failed to update lock: leases.coordination.k8s.io "gitlab-controller.knative.dev.eventing-gitlab.pkg.reconciler.source.reconciler.00-of-01" is forbidden: User "system:serviceaccount:knative-sources:gitlab-controller-manager" cannot update resource "leases" in API group "coordination.k8s.io" in the namespace "knative-sources"
````

Reply:
I have updated the original post accordingly.  Thanks for your patience.

Reply:
Thanks! Makes more sense now :) And nice catch 👏 

To fix this issue we should instead update the lease name, because there is really no reason to give the controller more permissions than it needs (like updating **all** leases in the namespace 🚨).

```diff
- gitlab-controller.knative.dev-eventing-gitlab-pkg-reconciler-source.reconciler.00-of-01
+ gitlab-controller.knative.dev.eventing-gitlab.pkg.reconciler.source.reconciler.00-of-01
```

(notice how dashes were changed to dots)

Reply:
I misunderstood the issue. I didn't realize that I was following a 'chmod 777' way in order to solve the issue :P. Thank you for your guidance.